bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/base/SEDFactory.java

```diff
@@ -143,10 +143,22 @@ private static VisitableSampledSpectrum getSED(final SourceDefinition sdp, final
         } else if (sdp.distribution() instanceof UserDefinedSpectrum) {
             final UserDefinedSpectrum userDefined = (UserDefinedSpectrum) sdp.distribution();
             final MagnitudeBand band = sdp.normBand();
-            // The user-supplied SED must cover the range of the instrument configuration and the normalization band.
+            Double inst_start_wave = instrument.getObservingStart();
```

Comment:
I've the impression that you could do this overriding `getObservingStart` in `Gnirs`

Reply:
Good point. It's not clear whether this special case always applies to GNIRS or only for this particular computation.

Reply:
The context is that previously we parsed and imported the entire user-supplied SED, but if the uploaded file covered a silly range it would trigger an OutOfMemoryError without giving any feedback to the user.  The implemented solution is to only import the data that are needed for the calculation.  This requires knowing the full wavelength range of the instrument configuration.  That is normally returned by instrument.getObservingStart/End.  However, some instruments override that method to use different algorithms for determining the range of wavelengths they are sensitive to (and that's fine), but GNIRS overrides those methods to return the wavelength range of each cross-dispersed order.  That is correct for most uses, but in this particular case, we need the *full* range over all orders.  This is defined by a single component (the XD filter), so I'm fine hard-coding it here if everybody else is. :-)

Reply:
👍  I think what you have here makes sense.

Reply:
👍